File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        s: str = str(x)
        i: int = 0
        j: int = len(s) - 1

        while i < j:
            if s[i] != s[j]:
                return False
            
            i += 1
            j -= 1

        return True

File: 0009-palindrome-number/0009-palindrome-number.go
func isPalindrome(x int) bool {
    var s string = strconv.Itoa(x)
    
    var l int = 0
    var r int = len(s) - 1
    
    for l < r {
        if s[l] != s[r] {
            return false
        }
        
        l += 1
        r -= 1
    }
    
    return true
}